src/Common/tests/System/Buffers/NativeMemoryManager.cs

```diff
@@ -54,7 +54,7 @@ public bool IsRetained
 
         public override unsafe MemoryHandle Pin(int elementIndex = 0)
         {
-            if (elementIndex < 0 || elementIndex > _length)
+            if (elementIndex < 0 || elementIndex >= _length)
```

Comment:
You can use `uint` to eliminate the explicit check for negative values, because a negative int casted to `uint` will give a value > int.MaxValue.

So
```c#
if ((uint)elementIndex >= (uint)_length)
```